modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js

```diff
@@ -86,11 +99,61 @@ class SegmentEdit extends Component {
 		);
 	}
 
-	_handleQueryChange = () => {
-		this.setState(
+	_fetchMembersCount = () => {
+		const formElement = document.getElementById(this.props.formId);
+
+		const formData = new FormData(formElement);
```

Comment:
Hmm, what about using formik for our form instances?

Reply:
Formik is used for validation purposes, but we're not using it as is intended to be used. This is because the Save button triggers the actual form, that comes with some extra information on the first render and inputs that react renders.

In this particular request, it's an aside request, the form doesn't need to be valid. But the request needs the actual information. It is far from perfect (IMO), but it works.

Reply:
I could use some advice on this. I just moved the method one level up as it was.

Reply:
Its probably fine to do things this way since you are passing in a `formId` so its fine to use `FormData` with that.